workers/data_refinery_workers/processors/create_compendia.py

```diff
@@ -119,132 +127,164 @@ def _prepare_frames(job_context: Dict) -> Dict:
-            if sum_val < rnaseq_tenth_percentile:
-                rows_to_filter.append(x)
 
-        del rnaseq_row_sums
+def _cached_remove_zeroes(job_context: Dict) -> Dict:
+    """
+    - Set all zero values in log2_rnaseq_matrix to NA, but make sure to keep track of
+      where these zeroes are
+    """
 
-        log_state("actually calling drop()", job_context["job"].id)
+    if job_context.get("log2_rnaseq_matrix") is None:
+        job_context["cached_zeroes"] = {}
+        return job_context
 
-        filtered_rnaseq_matrix = job_context.pop("rnaseq_matrix").drop(rows_to_filter)
+    cache_start = log_state("start caching zeroes", job_context["job"].id)
 
-        del rows_to_filter
+    # Cache our RNA-Seq zero values
+    cached_zeroes = {}
+    for column in job_context["log2_rnaseq_matrix"].columns:
+        cached_zeroes[column] = job_context["log2_rnaseq_matrix"].index[
+            np.where(job_context["log2_rnaseq_matrix"][column] == 0)
+        ]
 
-        log_state("end drop all rows", job_context["job"].id, drop_start)
-        log2_start = log_state("start log2", job_context["job"].id)
+    # Set all zero values in log2_rnaseq_matrix to NA, but make sure
+    # to keep track of where these zeroes are
+    job_context["log2_rnaseq_matrix"][job_context["log2_rnaseq_matrix"] == 0] = np.nan
 
-        # log2(x + 1) transform filtered_rnaseq_matrix; this is now log2_rnaseq_matrix
-        filtered_rnaseq_matrix_plus_one = filtered_rnaseq_matrix + 1
-        log2_rnaseq_matrix = np.log2(filtered_rnaseq_matrix_plus_one)
-        del filtered_rnaseq_matrix_plus_one
-        del filtered_rnaseq_matrix
+    log_state("end caching zeroes", job_context["job"].id, cache_start)
 
-        log_state("end log2", job_context["job"].id, log2_start)
-        cache_start = log_state("start caching zeroes", job_context["job"].id)
+    job_context["cached_zeroes"] = cached_zeroes
+
+    return job_context
 
-        # Cache our RNA-Seq zero values
-        cached_zeroes = {}
-        for column in log2_rnaseq_matrix.columns:
-            cached_zeroes[column] = log2_rnaseq_matrix.index[
-                np.where(log2_rnaseq_matrix[column] == 0)
-            ]
 
-        # Set all zero values in log2_rnaseq_matrix to NA, but make sure
-        # to keep track of where these zeroes are
-        log2_rnaseq_matrix[log2_rnaseq_matrix == 0] = np.nan
+def _full_outer_join_gene_matrices(job_context: Dict) -> Dict:
+    """
+    - Perform a full outer join of microarray_expression_matrix and
+      log2_rnaseq_matrix; combined_matrix
+    """
 
-        log_state("end caching zeroes", job_context["job"].id, cache_start)
+    # We potentially can have a microarray-only compendia but not a RNASeq-only
+    # compendia, so we need to provide a default
+    log2_rnaseq_matrix = job_context.pop("log2_rnaseq_matrix", None)
 
     outer_merge_start = log_state("start outer merge", job_context["job"].id)
```

Comment:
Can we actually move this into the `if` statement so we only log about it if we actually do it?